apl/data-types/map-fields.mdx

```diff
@@ -0,0 +1,34 @@
+---
+title: 'Map fields'
+description: 'This page explains what map fields are and how to query them.'
+tags: ['axiom documentation', 'documentation', 'axiom', 'map fields', 'object fields', 'data types']
+---
+
+Map fields are a special type of field that can hold a collection of nested key-value pairs within a single field. You can think of the content of a map field as a JSON object.
+
+## Benefits and drawbacks of map fields
+
+The benefit of map fields is that you can store additional attributes without adding more fields. This is particularly useful when the shape of your data is unpredictable (for example, additional attributes added by OpenTelemetry instrumentation). Using map fields means that you can avoid reaching the field limit of a dataset and improve query performance.
+
+The drawback of map fields is that querying them uses more query-hours than querying conventional fields. Additionally, you don’t have visibility into map fields from the schema. For example, autocomplete doesn’t know the properties inside the map field.
+
+## Custom attributes in tracing datasets
+
+If you use [OpenTelemetry](/send-data/opentelemetry) to send data to Axiom, you find some attributes in the `attributes.custom` map field. The reason is that instrumentation libraries can add hundreds or even thousands of arbitrary attributes to spans. Storing each custom attribute in a separate field would significantly increase the number of fields in your dataset. To keep the number of fields in your dataset under control, Axiom places all custom attributes in the single `attributes.custom` map field.
+
+## Use map fields in queries
+
+The example query below uses the `http.protocol` property inside the `attributes.custom` map field to filter results:
```

Comment:
Should also add an example about accessing properties of nested maps (`where ['map.field'].prop.foo == 14`)